awscli/clidocs.py

```diff
@@ -159,6 +159,15 @@ def doc_options_start(self, help_command, **kwargs):
     def doc_option(self, arg_name, help_command, **kwargs):
         doc = help_command.doc
         argument = help_command.arg_table[arg_name]
+        if argument.cli_name == '--generate-cli-skeleton':
+            doc.style.start_note()
+            msg = ("AWS CLI version 2's implementation of "
+                   "``generate-cli-skeleton`` selects the first element "
+                   "if an input shape contains an enum. This behavior "
+                   "may not be backwards-compatible with version 1's "
+                   "implementation, which selects a random element.")
```

Comment:
For the message that we are adding let's:

1. Not make it a note directive especially if we are adding it to the docstring. Mainly, I want to make sure we do not have too many note blocks in the documentation as it becomes hard to read.
2. For the content, let's just append to the end of the current docstring: `The generated JSON skeleton is not stable between versions of the AWS CLI and there are no backwards compatibility guarantees in the JSON skeleton generated.` I'm mainly suggesting this because the note about the enum is a bit specific and in reality, we may change more than just the enum values (e.g. we may change the placeholder values too in the future) and it gets more to the point that users should be carefully in using the value programatically.